syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.api";
option java_outer_classname = "Query";
option objc_class_prefix = "HLW";

package query;

service QueryService {
    rpc GetAllVesselIds(AllVesselIdsRequest) returns (AllVesselIdsResponse) {}
    rpc GetVesselIMU(IMURequest) returns (IMUResponse) {}    
    rpc GetVesselBounds(VesselBoundsRequest) returns (VesselBoundsResponse) {}
}

// Basic types
message Vec3 {
    float x = 1;
    float y = 2;
    float z = 3;
}

// Bounds messages
message Bounds {
    Vec3 center = 1;
    Vec3 extents = 2;
    Vec3 max = 3;
    Vec3 min = 4;
    Vec3 size = 5;
}

message VesselBoundsRequest {
    string vesselId = 1; 
}

message VesselBoundsResponse {
    Bounds bounds = 1;
}

// Vessel messages
message AllVesselIdsRequest {
   string type = 1; 
}

message AllVesselIdsResponse {
    repeated string ids = 1;
}

// IMU messages
message IMURequest {
    string vesselId = 1;
}

message IMUResponse {

    Vec3 position = 1; 
    Vec3 velocity = 2;
    Vec3 angularVelocity = 3;
    Vec3 angle = 4;
}
